packages/arb-node-core/nodehealth/nodehealth.go

```diff
@@ -124,16 +134,19 @@ func (config *configStruct) loadConfig() {
 	config.primaryHealthcheckRPC = ""
 	config.successCode = defaultSuccessCode
 	config.blockDifferenceTolerance = defaultBlockDifferenceTolerance
+
+	return &config
 }
 
 func checkEndpoint(config *configStruct, endpoint string) healthcheck.Check {
```

Comment:
Strings are copied by value so as is this won't change if the original config variable changes. If we want that, we need endpoint to be a pointer or to read from configStruct directly